Sources/TuistKit/Utils/XCResultParser.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import XCResultKit
+
+final class XCResultParser {
```

Comment:
```suggestion
struct XCResultParser: XCResultParsing {
```

Let's make it a struct and conform to a protocol, so we can dependency inject it. That way, we'll be also able to get rid of the `.xcresult` fixture. We don't need to test that `XCResultParser` is working as we're using a dependency for that.

I'll also add acceptance tests for the proprietary pieces of code, so we test the whole functionality end-to-end.